test/hotspot/jtreg/serviceability/jvmti/vthread/PopFrameTest/PopFrameTest.java

```diff
@@ -145,6 +145,7 @@ public static void run() {
             }
             log("Main #B.2: got expected JVMTI_ERROR_NONE");
             resumeThread(testTaskThread);
+            ensureAtBreakpoint();
```

Comment:
Not sure if it should work. 
Thevariable  `bp_sync_reached` is reset in notifyAtBreakpoint(). 
I see the previous call to  ensureAtBreakpoint(); in line 131. So it means that 
bp_sync_reached is true in this line and never rest after it between 131 and 148.
So I expect that this check in 148 is just return always without waiting anything. So it is not clear for me what is expected here.

Reply:
Nice catch, thanks.
Added one more call to `notifyAtBreakpoint()` to reset the state variable `bp_sync_reached`.